core/src/main/java/tech/pegasys/web3signer/core/util/OpenApiSpecsExtractor.java

```diff
@@ -0,0 +1,206 @@
+                fixRelativePathInOpenApiMap(path, yamlMap);
+                // write map back as yaml
+                objectMapper.writerWithDefaultPrettyPrinter().writeValue(path.toFile(), yamlMap);
+
+              } catch (final IOException e) {
+                throw new UncheckedIOException(e);
+              }
+            });
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  private void fixRelativePathInOpenApiMap(final Path yamlFile, final Map<String, Object> yamlMap) {
+    for (Map.Entry<String, Object> entry : yamlMap.entrySet()) {
+      final String key = entry.getKey();
+      final Object value = entry.getValue();
+
+      if ("$ref".equals(key) && value instanceof String) {
+        modifyRefValue(yamlFile, entry);
+      } else if ("discriminator".equals(key) && value instanceof Map) {
+        if (((Map<String, Object>) value).containsKey("mapping")) {
+          Map<String, Object> discriminatorMappings =
+              (Map<String, Object>) ((Map<String, Object>) value).get("mapping");
+          discriminatorMappings
+              .entrySet()
+              .forEach(discriminatorMapEntry -> modifyRefValue(yamlFile, discriminatorMapEntry));
+        }
+      } else if (value instanceof List) {
+        List listItems = (List) value;
+        for (Object listItem : listItems) {
+          if (listItem instanceof Map) {
+            fixRelativePathInOpenApiMap(yamlFile, (Map<String, Object>) listItem);
+          } // else we are not interested
+        }
+      } else if (value instanceof Map) {
+        fixRelativePathInOpenApiMap(yamlFile, (Map<String, Object>) value);
+      }
+    }
+  }
+
+  private void modifyRefValue(final Path yamlFile, final Map.Entry<String, Object> entry) {
+    final String fileName = StringUtils.substringBefore(entry.getValue().toString(), "#");
+    final String jsonPointer = StringUtils.substringAfter(entry.getValue().toString(), "#");
+    if (StringUtils.isNotBlank(fileName)) {
+      // convert filename path to absolute path
+      final Path parent = yamlFile.getParent();
+      final Path nonNormalizedRefPath = parent.resolve(Path.of(fileName));
+      // remove any . or .. from path
+      final Path normalizedPath = nonNormalizedRefPath.toAbsolutePath().normalize();
+
+      final String updatedValue =
+          StringUtils.isBlank(jsonPointer)
+              ? normalizedPath.toString()
+              : normalizedPath + "#" + jsonPointer;
+      entry.setValue(updatedValue);
+    }
+  }
+
+  private List<Path> extractToDestination() throws IOException {
+    final List<Path> extractedResources = new ArrayList<>();
+
+    try (final Stream<URL> webrootYamlFiles =
+        Resources.find("/" + OPENAPI_RESOURCES_ROOT + "**.*")) {
+      webrootYamlFiles.forEach(
+          openapiResource -> {
+            final String jarPath = openapiResource.getPath();
+            // the path is in file://.jar!<path> format. We need the non-jar path.
+            final String filePath =
```

Comment:
This is going to fail if we run Web3Signer through our IDE. It was failing locally for me

Reply:
fixed.